src/fetch.js

```diff
@@ -0,0 +1,13 @@
+const GlobalContext = require('./GlobalContext');
+const fetch = require('window-fetch');
+
+Object.assign(module.exports, fetch);
```

Comment:
This isn't the usual way we export in the rest of the codebase. It's usually

```
module.exports.fetch = fetch;
```

Reply:
`src/fetch.js` needs to export everything that `window-fetch` exports, including `Header`, `Request`, `Response`, and `Blob`. This accomplishes that.

The idea is that `src/fetch.js` is a drop-in replacement for `window-fetch`.

Reply:
If that's the case, maybe it's best this is put upstream into `window-fetch` as an interception API?

Reply:
Mm, I would say that overriding the `User-Agent` header and forcing it to be a specific value is a browser-specific concern. It might be nice if `window-fetch` provided a way to do that, but the tradeoff is that the API becomes more complicated.

Reply:
I think that's cleaner than requiring a hack to collect and re-export properties from the module.

`window-fetch` is designed to implement the needs of a "browser", and setting a User-Agent is one of the spec requirements of that.

Reply:
It does provide a way to do that: by passing in a `{header: {['user-agent']: ...}}` value as an argument to `fetch`.

Reply:
It's not really a hack. It's the equivalent of `export * from 'window-fetch';`

Reply:
The API I'm proposing isn't the ability to pass in a header, but to remember a header for future invocations. It's similar to this closure, but without the need for any re-exporting.

Reply:
See 'aggregating modules' in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/export

Reply:
The functionality is fine, but it adds a very unconventional (for the current codebase) new export style, necessitated mostly because the API isn't being defined in the right place.

If this is all contained in `window-fetch` as originally proposed, `fetch.js` itself would not need to exist, nor would this be a concern future engineers need to worry about getting wrong.

Reply:
It's not sufficient to remember the header. `fetch` itself must examine the incoming options on every invocation, and specifically set `User-Agent` to the standard exokit user agent string.

Otherwise, userland javascript would be able to set their own User-Agent string by calling `fetch(url, {header: {'user-agent': ...}})`, which is a security violation.

Reply:
> It's not sufficient to remember the header. fetch itself must examine the incoming options on every invocation, and specifically set User-Agent to the standard exokit user agent string.

This can be implemented entirely in `window-fetch` as proposed.

> Otherwise, userland javascript would be able to set their own User-Agent string by calling fetch(url, {header: {'user-agent': ...}})

This is fine -- Exokit allows user code to spoof.

Reply:
It's *very* surprising as a user of `window-fetch` that *any other module* can do `fetch = require('window-fetch'); fetch.forceTheUserAgentStringToBe(x);`

That means if you add a future dependency from an untrusted author, that author can easily publish an updated package which reconfigures `window-fetch` with a malicious `User-Agent` string.

Reply:
Malicious modules are a valid concern, but a malicious module could do much worse than setting a User-Agent -- and it's not clear that that itself would be a problem regardless.

The API wouldn't necessarily need to be `forceTheUserAgentStringToBe` -- it could be options on a constructor on the import. This would be roughly equivalent to `fetch.js`, except owned by `window-fetch` instead of Exokit.

Reply:
> This is fine -- Exokit allows user code to spoof.

Chrome does not: https://developer.chrome.com/multidevice/user-agent

You can reconfigure the user-agent string via extensions (https://chrome.google.com/webstore/detail/user-agent-switcher-for-c/djflhoibgkdhkhhcedjiklpkjnoahfmg?hl=en-US) but it's switched globally; there's no such thing as overriding it on a per-request basis.

This is important to prevent userland JS from spoofing other devices. Google Analytics also relies on standard User-Agent values, so allowing users to override it can break GA.

Reply:
Those concerns are outside the scope of Exokit. Exokit is not a browser.

Reply:
Fair enough. In that case, feel free to close this PR. I'll merge it into [sweetiekit-dom](https://github.com/sweetiebird/sweetiekit-dom) which is what we bundle with [SweetieKit](https://github.com/sweetiebird/sweetiekit).

That will fix our GA issue #1230 

I was hoping to maintain parity with exokit's DOM implementation, but it sounds like it might take some time before the necessary functionality / fixes can be integrated into `window-fetch` and published. The `window-fetch` extensions also introduce risks due to the dependency on mutable global state.

It might be a better idea for us to switch over to the standard https://github.com/bitinn/node-fetch module. There are significant upstream differences, like support for `stream.pipe`. 
![image](https://user-images.githubusercontent.com/59632/60199125-db6c9f00-9808-11e9-97be-43e4374285eb.png)

Reply:
> sounds like it might take some time before the necessary functionality / fixes can be integrated into window-fetch and published.

Which timing are does this refer to? It looks like most of that work is done in this PR.

> The window-fetch extensions also introduce risks due to the dependency on mutable global state.

What are the risks?

> There are significant upstream differences, like support for stream.pipe.

Indeed, `window-fetch` should pull from upstream.